### ISSUE TITLE
Different transition speed when using `n%` and `+n%`

### ISSUE THREAD

**Issue by max-baz** [geekylthyosaur/slight#8]
Starting both examples with `0%` brightness, and printing the value [here](https://github.com/geekylthyosaur/slight/blob/master/src/device.rs?rgh-link-date=2023-07-31T07%3A03%3A54Z#L30) instead of changing it as you suggested results in the output below.

`sys/class/backlight/apple-panel-bl/max_brightness` contains `500`.

```
$ ./target/debug/slight '50%'
1
2
3
4
5
6
7
8
9
10
11
...
248
249
250

$ ./target/debug/slight '+50%'
5
10
15
20
24
30
35
40
44
49
55
...
240
244
250
```

**Comment by geekylthyosaur** [comment]
Oh I missed the `+` part in https://github.com/geekylthyosaur/slight/issues/5#issuecomment-1657282198.

It works like that because I've made it so `+-n%` is [the only mode](https://github.com/geekylthyosaur/slight/blob/master/src/range.rs#L88) that can use exponential range so, if no `--exponent` was provided, it uses 1.0 by default, thus it produces range with step 5 because of some math magic.

`n%` (and all other inputs) on the other hand use the `curr..=new` range and iterate over each value, so it takes more time.

It should be fixed by adding `.step_by()` [here](https://github.com/geekylthyosaur/slight/blob/master/src/range.rs#L39-L40). But it requires some math to choose good step value to make it work for different values of `max_brightness`.

I'll take a look soon.

**Comment by geekylthyosaur** [comment]
So I've implemented d0ce59e which adds `--max-iter` with default value 10. This means that the brightness will always be set in at most 10 (or the given value) iteration, in constant time.

Take a look. If you have any ideas to improve it, I would be happy to hear them.

**Comment by max-baz** [comment]
I think it's perfect! Thanks :+1:
